client/xmlrpc_client_test.go

```diff
@@ -66,3 +71,56 @@ func TestExecuteCall(t *testing.T) {
 		})
 	}
 }
+
+func TestExecuteCallWithTimeout(t *testing.T) {
+	tt := []struct {
+		name             string
+		connectTimeout   int
+		requestTimeout   int
+		sleepTime        time.Duration
+		methodName       string
+		args             []interface{}
+		expectedResponse map[string]interface{}
+		expectedError    string
+	}{
+		{name: "ExecuteCall Success",
+			connectTimeout: 2,
+			requestTimeout: 2,
+			sleepTime:      1,
+		},
+		{name: "ExecuteCall Fail Timeout",
+			connectTimeout: 2,
+			requestTimeout: 2,
+			sleepTime:      4,
+			expectedError:  "i/o timeout",
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			//init server
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				log.Println("Sleeping...")
+				time.Sleep(tc.sleepTime * time.Second)
+				io.WriteString(w, `response...`)
+			}))
+			defer ts.Close()
+
+			//init client
+			client := NewClient(tc.connectTimeout, tc.requestTimeout)
+
+			response, err := client.ExecuteCall(ts.URL, "test", []interface{}{})
+
+			//If timeout happens
+			if err != nil && !strings.Contains(err.Error(), tc.expectedError) {
+				t.Fatalf("expected and actual doesn't match, Actuval was: %v, Expected was: %v", err.Error(), tc.expectedError)
+			}
+
+			// if timeout doesn't appear, we expect a response
+			if err == nil && response == nil {
+				t.Fatalf("Respone was expected but came nothing")
+			}
```

Comment:
shouldn't we also check if the response is the expected? if not, we should remove the 'expectedResponse' field form the test table struct